cwltool/pathmapper.py

```diff
@@ -139,6 +144,33 @@ def trim_listing(obj):
     if obj.get("location", "").startswith("file://") and "listing" in obj:
         del obj["listing"]
 
+# Download http Files
+def downloadHttpFile(httpurl):
+    # type: (Text) -> Text
+    cache_session = None
+    if "XDG_CACHE_HOME" in os.environ:
+        cache_session = CacheControl(
+            requests.Session(),
+            cache=FileCache(
+                os.path.join(os.environ["XDG_CACHE_HOME"], "cwltool")))
+    elif "HOME" in os.environ:
+        cache_session = CacheControl(
+            requests.Session(),
+            cache=FileCache(
+                os.path.join(os.environ["HOME"], ".cache", "cwltool")))
+    else:
+        cache_session = CacheControl(
+            requests.Session(),
+            cache=FileCache(
+                os.path.join(os.path.expanduser('~'), ".cache", "cwltool")))
```

Comment:
Can you refactor this so the if/elif/else is just determining the directory to put `.cache` in and the call to CacheControl occurs afterwards?